neo/Consensus/ConsensusService.cs

```diff
@@ -77,7 +78,17 @@ private bool AddTransaction(Transaction tx, bool verify)
             {
                 // if we are the primary for this view, but acting as a backup because we recovered our own
                 // previously sent prepare request, then we don't want to send a prepare response.
-                if (context.IsPrimary || context.WatchOnly) return true;
+                if (context.IsPrimary || context.WatchOnly) return true;
+
+                // Check policy
+                var block = context.CreateBlock();
+
+                if (block.Size > NativeContract.Policy.GetMaxBlockSize(context.Snapshot))

```

Comment:
Looks good, this var block is local, right?
Can it be just `context.CreateBlock().Size` in the if?